java-checks-test-sources/src/main/java/checks/BoxedBooleanExpressionsCheck.java

```diff
@@ -341,4 +341,72 @@ Boolean False() {
 
   Object foo() { return new Object(); }
   Object bar() { return new Object(); }
+
```

Comment:
I will also suggest adding more test cases:

- where condition is in else branch

-  where there are other usages of checked variable

- when the tested variable is actually null

-  etc.